### PR TITLE
Meeting side panel deep link - C# and NodeJS

### PR DESCRIPTION
**Pull request by HunaidHanfee-MSFT**

### CODE REVIEW

File: samples/tab-deeplink/csharp/DeepLinkBot/Manifest/manifest.json [hunk 1]
```diff
@@ -30,6 +30,14 @@
       "scopes": [
         "groupchat",
         "team"
+      ],
+      "context": [
+        "channelTab",
+        "privateChatTab",
+        "meetingSidePanel",
+        "meetingStage",
+        "meetingChatTab",
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Manifest/manifest.json:39]: Do we need all of these context ?

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml [hunk 2]
```diff
@@ -8,9 +8,50 @@
   crossorigin="anonymous"
 ></script>
 <script>
+var deepLinkString = "";
+
+function callSidepanelDeeplink() {
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml:13]: callSidePanelDeeplink

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml [hunk 3]
```diff
@@ -60,10 +101,58 @@
...
+            Extended Deep link features
+        </span>
+        @{
+            Html.RenderPartial("ExtendedDeepLinks");
+        }
+    </div>
+
+    <div id="side-panel-content" class="container" style="color:white;display:none;margin-top:3rem;">
+        <div>
+            This is a side panel tab.
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml:116]: Side Panel

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml [hunk 4]
```diff
@@ -74,6 +74,17 @@
 
 <div>
     <div class="container">
+        <div class="card-container-div">
+            <div>
+                <div class="card" style="min-height:6rem;width:45rem;">
+                    <div class="card-body">
+                        <div class="card-header">
+                            <b>Add this app in meeting and stage this tab to experience deeplink to sidepanel.</b>
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:82]: meeting side panel

File: samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html [hunk 5]
```diff
@@ -1,27 +1,105 @@
...
+                        </div>
+                        <div class="configure-header">
+                            To use this, configure the deeplink with following parameters
+                            <!-- Since we are using pre tag for having the code snippet in specified format ignoring the indentation here -->
+                            <pre style="overflow: hidden;">
+microsoftTeams.app.openLink("https://teams.microsoft.com/l/entity/{appId}/{entityId}?context={"chatId": "{meetingId}","contextType":"chat"}")
+                            </pre>
+                        </div>
+                        <div>Click on button to open meeting sidepanel.</div>
+                    </div>
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html:89]: side panel

File: samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html [hunk 6]
```diff
@@ -1,27 +1,105 @@
...
+                    <button 
+                        data-id="default"
+                        class="callDeeplinkButton"
+                        onclick="callSidepanelDeeplink()"
+                        style="width:120px"
+                    >
+                        Open side panel
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html:96]: Open Teams meeting side panel

File: samples/tab-deeplink/nodejs/pages/ExtendedDeepLink.html [hunk 7]
```diff
@@ -77,6 +77,17 @@
 
 <div>
     <div class="container">
+        <div class="card-container-div">
+            <div>
+                <div class="card" style="min-height:6rem;width:45rem;">
+                    <div class="card-body">
+                        <div class="card-header">
+                            <b>Add this app in meeting and stage this tab to experience deeplink to sidepanel.</b>
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/nodejs/pages/ExtendedDeepLink.html:85]: meeting side panel

File: samples/tab-deeplink/nodejs/teamsAppManifest/manifest.json [hunk 8]
```diff
@@ -30,6 +30,14 @@
       "scopes": [
         "groupchat",
         "team"
+      ],
+      "context": [
+        "channelTab",
+        "privateChatTab",
+        "meetingSidePanel",
+        "meetingStage",
+        "meetingChatTab",
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/nodejs/teamsAppManifest/manifest.json:39]: Check if these all context is needed.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Reviewed and added my comments.

File: samples/tab-deeplink/csharp/DeepLinkBot/Manifest/manifest.json (hunk 1 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/csharp/DeepLinkBot/Manifest/manifest.json:39]: Yes, only one we can remove but I keep it, so the user doesn't get confused about app installation.

File: samples/tab-deeplink/nodejs/teamsAppManifest/manifest.json (hunk 8 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/nodejs/teamsAppManifest/manifest.json:39]: Yes. needed

File: samples/tab-deeplink/nodejs/pages/ExtendedDeepLink.html (hunk 7 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/nodejs/pages/ExtendedDeepLink.html:85]: Done

File: samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html (hunk 6 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html:96]: Done

File: samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html (hunk 5 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/nodejs/pages/ChannelDeepLink.html:89]: Done

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml (hunk 4 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/ExtendedDeepLinks.cshtml:82]: Done

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml (hunk 3 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml:116]: Done

File: samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml (hunk 2 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/csharp/DeepLinkBot/Views/Home/DeepLinkChannel.cshtml]: Done

File: samples/tab-deeplink/nodejs/README.md [hunk 9]
```diff
@@ -93,3 +93,11 @@ Enter text in the emulator.  The text will be echoed back by the bot.
 
 ![Additional features](Images/DeepLinkTab2.png)
```
Comment (ChetanSharma-msft) [inline samples/tab-deeplink/nodejs/README.md:94]: We can hide app id?
image

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added a single comment

File: samples/tab-deeplink/nodejs/README.md (hunk 9 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-deeplink/nodejs/README.md:94]: Its Polly app Id, that's public for all the Teams admins so I don't think it will cause any issue

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**